specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json

```diff
@@ -114,6 +153,185 @@
+          "items": {
+            "type": "string"
+          }
+        },
+        "NodeTypes": {
+          "description": "The list of node types.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "ClusterVersions": {
+          "description": "The list of cluster versions.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "vmsizes": {
+          "description": "The list of virtual machine sizes.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "RegionalQuotaCapability": {
+      "description": "The regional quota capacity.",
+      "properties": {
+        "region_name": {
+          "description": "The region name.",
+          "type": "string"
+        },
+        "cores_used": {
+          "description": "The number of cores used in the region.",
+          "type": "integer",
+          "format": "int64"
+        },
+        "cores_available": {
+          "description": "The number of cores available in the region.",
+          "type": "integer",
+          "format": "int64"
+        }
+      }
+    },
+    "QuotaCapability": {
+      "description": "The regional quota capability.",
+      "properties": {
+        "regionalQuotas": {
+          "description": "The list of region quota capabilities.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RegionalQuotaCapability"
+          }
+        }
+      }
+    },
+    "CapabilitiesResult": {
+      "description": "The Get Capabilities operation response.",
+      "properties": {
+        "versions": {
+          "description": "The version capability.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/VersionsCapability"
+          }
+        },
+        "regions": {
+          "description": "The virtual machine size compatibility features.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/RegionsCapability"
+          }
+        },
+        "vmSizes": {
+          "description": "The virtual machine sizes.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/VmSizesCapability"
+          }
+        },
+        "vmSize_filters": {
+          "description": "The virtual machine size compatibility filters.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VmSizeCompatibilityFilter"
+          }
+        },
+        "features": {
+          "description": "The capability features.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "quota": {
+          "description": "The quota capability.",
+          "$ref": "#/definitions/QuotaCapability"
+        }
+      }
+    },
```

Comment:
I guess all properties of `CapabilitiesResult` model and the model it composes should be readonly, could you double check?

Reply:
> I guess all properties of `CapabilitiesResult` model and the model it composes should be readonly, could you double check?

Hi anuchandy, I have added the readonly parameter